app/scripts/views/components/dataTable/tableViewTemplate.js

```diff
@@ -48,20 +51,27 @@
     },
     events: {
       'show-loader': function() {
-        this.showLoad = true;
+        if (!this.madeSelection) {
+          this.showLoad = true;
+        }
       },
       'gene-list-updated': function(genes) {
         genes = $.extend(true, [], genes);
         this.chartInst.updateGenes(genes);
       },
       'update-special-charts': function() {
-        var attrId =
-          this.attributes.group_type === 'patient' ? 'patient_id' : 'sample_id';
-        var _selectedCases =
-          _.pluck(this.invisibleDimension.top(Infinity), attrId);
-        this.chartInst.update(_selectedCases, this.selectedRows);
-        this.setDisplayTitle(this.chartInst.getCases().length);
-        this.showLoad = false;
+        // Do not update chart if the selection is made on itself
```

Comment:
I remember that we though of updating special tables(Mutated genes and CNA tables) on submit operation (this is basically for updating the Freq column). May be we need to discuss about this.